prov/util/src/util_cmap.c

```diff
@@ -571,12 +581,31 @@ struct util_cmap *ofi_cmap_alloc(struct util_ep *ep,
 		cmap->acquire = ofi_fastlock_acquire;
 		cmap->release = ofi_fastlock_release;
 	}
-	return cmap;
+
+	ep->cmap = cmap;
+
+	if (pthread_create(&cmap->cm_thread, 0,
+			   cmap->attr.cm_thread_func, ep)) {
+		FI_WARN(ep->av->prov, FI_LOG_FABRIC,
+			"Unable to create cmap thread\n");
+		ret = -FI_ENOMEM;
+		goto err3;
+	}
+
+	assert(ep->av);
+	ret = ofi_cmap_bind_to_av(cmap, ep->av);
+	if (ret)
+		goto err4;
+
+	return FI_SUCCESS;
+err4:
+	util_cmap_cm_thread_close(cmap);
 err3:
 	free(cmap->attr.name);
 err2:
 	free(cmap->handles_av);
 err1:
 	free(cmap);
-	return NULL;
+	ep->cmap = NULL;
+	return ret;
 }
```

Comment:
Can you add some verbal description to the commit message? Something along the lines of:
"If addresses were inserted into the AV before an endpoint is bound to the AV, cmap->handles_av would not be initialized. The patch fixes the issue by calling ofi_cmap_update (which initializes cmap->handles_av[fi_addr]) for every inserted address in the AV when cmap is created."